### PR TITLE
Feature/small refactoring

### PR DESCRIPTION
**Pull request by KeesCBakker**
Some small refactoring:

- adding of a dev container to equalize bash support. Mitigates #175 .
- dropped tests for node 16 as security support ended 7 months ago (11 Sep 2023), according to https://endoflife.date/nodejs
- audit fix update for packages.
- `parseInt` of the width and the height that come in as env variables.
- fixed test `should respond with a png graph in the default s3 region` failing half of the time on my machine by making the `panelId` variable.
- make testing more resilient by adding `--forbid-only --forbid-pending` to the GitHub workflow.

### CODE REVIEW

**Comment by KeesCBakker** [comment]
Made the testing faster in the dev container by using a volume (https://code.visualstudio.com/remote/advancedcontainers/improve-performance#_use-a-targeted-named-volume).

Test `should respond with a png graph in the default s3 region` fails half of the time (at my machine) because it sometimes returns panel 3 and sometimes 8. As we do not really care about the panel ID here (we're testing S3), I deduce the panel from the response.

```
    it('should respond with a png graph in the default s3 region', async () => {
      let response = await ctx.sendAndWaitForResponse('@hubot graf db AAy9r_bmk:cpu server=ww3.example.com now-6h');
      response = response.replace(/\/[a-f0-9]{40}\.png/i, '/abdcdef0123456789.png');

      let panelRegex = /panelId=(\d+)/
      expect(response).to.match(panelRegex);

      let panelId = response.match(panelRegex)[1]
      expect(response).to.eql(
        'CPU: https://graf.s3.us-standard.amazonaws.com/grafana/abdcdef0123456789.png - https://play.grafana.org/d/AAy9r_bmk/?panelId=' + panelId + '&fullscreen&from=now-6h&to=now&var-server=ww3.example.com'
      );
    });
```

Now it always works.

**Comment by KeesCBakker** [comment]
@stephenyeargin think this branch is ready to ship. Please check if you want to drop support for `node 16` and you like the way we've refactored the testing and bootstrapping.

File: src/grafana.js [hunk 1]
```diff
@@ -91,8 +91,8 @@ module.exports = (robot) => {
     let apiPanelId = false;
     let pname = false;
     const query = {
-      width: process.env.HUBOT_GRAFANA_DEFAULT_WIDTH || 1000,
-      height: process.env.HUBOT_GRAFANA_DEFAULT_HEIGHT || 500,
+      width: parseInt(process.env.HUBOT_GRAFANA_DEFAULT_WIDTH) || 1000,
+      height: parseInt(process.env.HUBOT_GRAFANA_DEFAULT_HEIGHT) || 500,
```
Comment (stephenyeargin) [inline src/grafana.js:95]: nit: My linter will complain about `parseInt()` if the second `radix` [parameter isn't defined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#description).

```suggestion
      width: parseInt(process.env.HUBOT_GRAFANA_DEFAULT_WIDTH, 10) || 1000,
      height: parseInt(process.env.HUBOT_GRAFANA_DEFAULT_HEIGHT, 10) || 500,
```

File: test/grafana-s3-test.js [hunk 2]
```diff
@@ -52,8 +52,12 @@ describe('s3', () => {
       let response = await ctx.sendAndWaitForResponse('@hubot graf db AAy9r_bmk:cpu server=ww3.example.com now-6h');
       response = response.replace(/\/[a-f0-9]{40}\.png/i, '/abdcdef0123456789.png');
 
+      let panelRegex = /panelId=(\d+)/
+      expect(response).to.match(panelRegex);
+
+      let panelId = response.match(panelRegex)[1]
       expect(response).to.eql(
-        'CPU: https://graf.s3.us-standard.amazonaws.com/grafana/abdcdef0123456789.png - https://play.grafana.org/d/AAy9r_bmk/?panelId=3&fullscreen&from=now-6h&to=now&var-server=ww3.example.com'
+        'CPU: https://graf.s3.us-standard.amazonaws.com/grafana/abdcdef0123456789.png - https://play.grafana.org/d/AAy9r_bmk/?panelId=' + panelId + '&fullscreen&from=now-6h&to=now&var-server=ww3.example.com'
```
Comment (stephenyeargin) [inline test/grafana-s3-test.js:60]: Thanks for tackling this one, it's been on my list for a while. I still don't know why it's an intermittent bug. I'd think with static data it'd be deterministic? Some sort of race condition?

**Review by stephenyeargin — APPROVED** [review]
LGTM. Left minor comments.

File: test/grafana-s3-test.js (hunk 2 above)
Comment (KeesCBakker) [reply test/grafana-s3-test.js:60]: Yeah, I have also no clue as to why it fails. With the current system of callbacks it is also really hard to determine why it fail.

**Pull request merged**